src/Watcher.cs

```diff
@@ -0,0 +1,100 @@
+﻿using System;
+using System.IO;
+using System.Runtime.Serialization;
+using System.Threading;
+using System.Threading.Tasks;
+using k8s.Exceptions;
+using Microsoft.Rest;
+using Microsoft.Rest.Serialization;
+
+namespace k8s
+{
+    public enum WatchEventType
+    {
+        [EnumMember(Value = "ADDED")] Added,
+
+        [EnumMember(Value = "MODIFIED")] Modified,
+
+        [EnumMember(Value = "DELETED")] Deleted,
+
+        [EnumMember(Value = "ERROR")] Error
+    }
+
+    public class Watcher<T> : IDisposable
+    {
+        private readonly StreamReader _streamReader;

```

Comment:
I don't think we use the underscore convention elsewhere, do we? I think we should be consistent.

Reply:
https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/coding-style.md

```
We use _camelCase for internal and private fields and use readonly where possibl
```

Actually, I use resharper out of box coding style formatter
https://www.jetbrains.com/resharper/features/code_formatting.html